src/Temporalio/Bridge/src/runtime.rs

```diff
@@ -137,6 +143,20 @@ pub extern "C" fn byte_array_free(runtime: *mut Runtime, bytes: *const ByteArray
 
 impl Runtime {
     fn new(options: &RuntimeOptions) -> anyhow::Result<Runtime> {
+        // Create custom meter here so it will be dropped on any error and
+        // therefore will call "free"
+        let custom_meter = unsafe {
+            options
+                .telemetry
+                .as_ref()
+                .map(|v| v.metrics.as_ref())
+                .flatten()
```

Comment:
map then flatten can usually be written as `and_then` assuming these end up as options

Reply:
Fixed